modules/caddypki/acmeserver/acmeserver.go

```diff
@@ -227,6 +259,62 @@ func (ash Handler) openDatabase() (*db.AuthDB, error) {
 	return database.(databaseCloser).DB, err
 }
 
+// makeClient creates an ACME client which will use a custom
+// resolver instead of net.DefaultResolver.
+func (ash Handler) makeClient() (acme.Client, error) {
+	for _, v := range ash.Resolvers {
+		addr, err := caddy.ParseNetworkAddress(v)
+		if err != nil {
+			// If a port wasn't specified for the resolver,
+			// try defaulting to 53 and parse again
+			if strings.Contains(err.Error(), "missing port in address") {
+				addr, err = caddy.ParseNetworkAddress(v + ":53")
+			}
+			if err != nil {
+				return nil, err
+			}
```

Comment:
I don't think this is right: a port is optional for `caddy.ParseNetworkAddress`, so it does not return an error, it just sets the port to 0.